classes/ObjectModel.php

```diff
@@ -887,6 +887,29 @@ public function deleteSelection($ids)
         return $result;
     }
 
+    /**
+     * Does a soft delete on current object, using the "deleted" field in DB
+     * If the model object has no "deleted" property or no "deleted" definition field it will throw an exception
+     *
+     * @return bool
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function softDelete()
+    {
+        $definitions = ObjectModel::getDefinition($this);
+
+        if (empty($definitions['fields']['deleted'])) {
+            throw new PrestaShopException('Field "deleted" is missing from definition in object model ' . get_class($this));
+        }
+        if (!array_key_exists('deleted', $this)) {
```

Comment:
Please don't test with array_key_exists for object :) It's not compatible with PHP7.4

Reply:
thanx, done!